### PR TITLE
executor: add warning for "drop user if exists"

### PR DESCRIPTION
**Pull request by july2993**
### What problem does this PR solve? 
fix https://github.com/pingcap/tidb/issues/10196
relate parser pr: https://github.com/pingcap/parser/pull/319

### What is changed and how it works?
if having `if exist`, return warnings for the not exists user, or return error.

the account is not escape, we need to finish this TODO
```go
// String converts UserIdentity to the format user@host.
func (user *UserIdentity) String() string {
    // TODO: Escape username and hostname.
    if user == nil {
        return ""
    }
    return fmt.Sprintf("%s@%s", user.Username, user.Hostname)
}
```

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
build TiDB and test

TiDB
```
mysql> DROP USER IF EXISTS idontexist;
Query OK, 0 rows affected, 1 warning (0.01 sec)

mysql> show warnings;
+-------+------+-----------------------------------+
| Level | Code | Message                           |
+-------+------+-----------------------------------+
| Note  | 3162 | User idontexist@% does not exist. |
+-------+------+-----------------------------------+
1 row in set (0.00 sec)

mysql>
```

mysql
```
Server version: 5.7.21-log Homebrew

mysql> DROP USER IF EXISTS idontexist;
Query OK, 0 rows affected, 1 warning (0.03 sec)

mysql> show warnings;
+-------+------+---------------------------------------+
| Level | Code | Message                               |
+-------+------+---------------------------------------+
| Note  | 3162 | User 'idontexist'@'%' does not exist. |
+-------+------+---------------------------------------+
1 row in set (0.00 sec)

mysql>
```

Code changes

Side effects

Related changes

### CODE REVIEW

File: infoschema/infoschema.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ var (
 	ErrTableExists = terror.ClassSchema.New(codeTableExists, "Table '%s' already exists")
 	// ErrTableDropExists returns for dropping a non-existent table.
 	ErrTableDropExists = terror.ClassSchema.New(codeBadTable, "Unknown table '%s'")
+	// ErrUserDropExists returns for dropping a non-existent user.
+	ErrUserDropExists = terror.ClassSchema.New(codeBadUser, "User %s does not exist.")
```
Comment (jackysp) [inline infoschema/infoschema.go]: I think it is better to define the error message in the parser repo.

**Comment by morgo** [comment]
LGTM. The related parser change has merged, please update go.sum.

File: infoschema/infoschema.go (hunk 1 above)
Comment (july2993) [reply infoschema/infoschema.go]: It seems the same kind error message all defined here, no any in parser repo, do you means put all this Err* in parser repo?

Comment (zz-jason) [reply infoschema/infoschema.go]: Hah, maybe we can use another PR to move all the error message in this file to the parser repo?

**Pull request merged**